OpenChange/MAPIStoreMailMessage.m

```diff
@@ -288,31 +360,69 @@ - (void) _fetchHeaderData
 
 - (void) _fetchBodyData
 {
-  NSData *rawContent;
-  NSString *resultKey;
-  id result;
-
   if (!headerSetup)
     [self _fetchHeaderData];
 
-  if (!bodyContent && mimeKey)
+  if (!bodyContent && bodyContentKeys)
     {
-      result = [sogoObject fetchParts: [NSArray arrayWithObject: mimeKey]];
+      id result;      
+      NSString *key;
+      NSEnumerator *enumerator;
+      NSMutableData *htmlContent = nil;
+      NSMutableData *textContent = nil;
+      
+      result = [sogoObject fetchParts: bodyContentKeys];
       result = [[result valueForKey: @"RawResponse"] objectForKey: @"fetch"];
-      if ([mimeKey hasPrefix: @"body.peek"])
-        resultKey = [NSString stringWithFormat: @"body[%@]",
-                              [mimeKey _strippedBodyKey]];
+
+      textContent = [[NSMutableData alloc] initWithCapacity: 0];
+      if (multipartMixed)
+        htmlContent = textContent;
       else
-        resultKey = mimeKey;
-      rawContent = [[result objectForKey: resultKey] objectForKey: @"data"];
-      ASSIGN (bodyContent, [rawContent bodyDataFromEncoding: headerEncoding]);
+        htmlContent = [[NSMutableData alloc] initWithCapacity: 0];      
+      
+      enumerator = [bodyContentKeys objectEnumerator];
+      while ((key = [enumerator nextObject]))
+        {
+          NSData *content = [[result objectForKey: key] objectForKey: @"data"];
+          NSString *encoding = [bodyPartsEncodings objectForKey: key];
+          NSString *mimeType = [bodyPartsMimeTypes objectForKey: key];
+          content = [content bodyDataFromEncoding: encoding];
+          if ([mimeType isEqualToString: @"text/html"])
+            {
```

Comment:
I know but I like to put it if the sentence to be executed spans more than one line, otherwise I get confused some times.
Should we make an exception for this cases?